### PR TITLE
IVM: Made description when creating an asset group required

### PR DESCRIPTION
**Pull request by jfrantz1-r7**
## Proposed Changes
Simply changed the boolean on `Description` under `Create Asset Group` `true` instead of `false`. 
### Description
The old action would fail if no description was set, but it was not a required field. Changed it to `true` so it is required.

Describe the proposed changes:

  - Changed the boolean to `true` for creating an asset group and regenerated the plugin

### Testing

Tested the plugin locally on my machine with an InsightVM instance. It looks like it doesn't validate on `make validate`:

`Input violations: Action -> "Create Site": Missing ['|included_addresses|[]string|[]|False|List of addresses to include in scan scope|None|', '|excluded_addresses|[]string|[]|False|List of addresses to exclude in scan scope|None|', '|included_asset_groups|[]integer|[]|False|Assets associated with these asset group IDs will be included in the site|None|', '|excluded_asset_groups|[]integer|[]|False|Assets associated with these asset group IDs will be excluded in the site|None|'] in help.md
Input violations: Action -> "Create Scan Engine": Missing ['|sites|[]integer|[]|False|List of site IDs with which to associate the engine|None|'] in help.md
Input violations: Action -> "Top Remediations": Missing ['|scope_ids|[]integer|[]|False|Scope IDs for which tickets should be generated, by default all are included|None|'] in help.md
Input violations: Action -> "Generate AdHoc SQL Report": Missing ['|scope_ids|[]integer|[]|False|Scope IDs for which tickets should be generated, by default all are included|None|'] in help.md
Input violations: Trigger -> "New Scans": Missing ["|status_filter|[]string|['Successful']|False|List of scan statuses to match for trigger; options include: Aborted, Successful, Running, Stopped, Failed, Paused, Unknown|None|"] in help.md
Input violations: Trigger -> "New Vulnerability Exception": Missing ["|status_filter|[]string|['Under Review']|False|List of vulnerability statuses to match against. Options include: Under Review and Approved|None|"] in help.md
`

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [✅  ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [✅  ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [✅  ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [✅  ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [✅  ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ✅ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [✅  ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [✅  ] Work fully completed
- [✅  ] Functional
  - [ ✅ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ✅ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [✅  ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ✅ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ✅ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by jschipp-r7** [comment]
@jfrantz1-r7 Hey Josh, if you grab the latest version of `icon-plugin` it should automatically update the markdown tables to pass validation. Easy way to grab the tooling is running `tools/update-tools.sh` from this repo, or using your package manager depending on how you installed icon-plugin.

File: rapid7_insightvm/help.md [hunk 1]
```diff
@@ -4167,68 +4167,26 @@ Example output:
...
 |Name|Type|Required|Description|
 |----|----|--------|-----------|
-|exception|vulnerability_exception|False|InsightVM Vulnerability Exception|
+|exception|vulnerability_exception|False|InsightVM vulnerability exception|
 
-Example output:
+Example Output:
 
 ```
-{
```
Comment (jschipp-r7) [inline rapid7_insightvm/help.md]: If this example output is valid, we should add it back.

**Comment by jfrantz1-r7** [comment]
@jschipp-r7 I ran `update-tools.sh` before regenerating this, i'll check again

**Comment by jfrantz1-r7** [comment]
@jschipp-r7 I made sure the newest icon-plugin tools were installed, and I'm still getting the same errors on validation:

`Input violations: Action -> "Create Site": Missing ['|included_addresses|[]string|[]|False|List of addresses to include in scan scope|None|', '|excluded_addresses|[]string|[]|False|List of addresses to exclude in scan scope|None|', '|included_asset_groups|[]integer|[]|False|Assets associated with these asset group IDs will be included in the site|None|', '|excluded_asset_groups|[]integer|[]|False|Assets associated with these asset group IDs will be excluded in the site|None|'] in help.md Input violations: Action -> "Create Scan Engine": Missing ['|sites|[]integer|[]|False|List of site IDs with which to associate the engine|None|'] in help.md Input violations: Action -> "Top Remediations": Missing ['|scope_ids|[]integer|[]|False|Scope IDs for which tickets should be generated, by default all are included|None|'] in help.md Input violations: Action -> "Generate AdHoc SQL Report": Missing ['|scope_ids|[]integer|[]|False|Scope IDs for which tickets should be generated, by default all are included|None|'] in help.md Input violations: Trigger -> "New Scans": Missing ["|status_filter|[]string|['Successful']|False|List of scan statuses to match for trigger; options include: Aborted, Successful, Running, Stopped, Failed, Paused, Unknown|None|"] in help.md Input violations: Trigger -> "New Vulnerability Exception": Missing ["|status_filter|[]string|['Under Review']|False|List of vulnerability statuses to match against. Options include: Under Review and Approved|None|"] in help.md`

**Comment by jschipp-r7** [comment]
This branch is now out of date since https://github.com/rapid7/insightconnect-plugins/pull/328 was merged today, will need to update it.

**Comment by gusingh-r7** [comment]
> @jschipp-r7 I made sure the newest icon-plugin tools were installed, and I'm still getting the same errors on validation:
> 
> `Input violations: Action -> "Create Site": Missing ['|included_addresses|[]string|[]|False|List of addresses to include in scan scope|None|', '|excluded_addresses|[]string|[]|False|List of addresses to exclude in scan scope|None|', '|included_asset_groups|[]integer|[]|False|Assets associated with these asset group IDs will be included in the site|None|', '|excluded_asset_groups|[]integer|[]|False|Assets associated with these asset group IDs will be excluded in the site|None|'] in help.md Input violations: Action -> "Create Scan Engine": Missing ['|sites|[]integer|[]|False|List of site IDs with which to associate the engine|None|'] in help.md Input violations: Action -> "Top Remediations": Missing ['|scope_ids|[]integer|[]|False|Scope IDs for which tickets should be generated, by default all are included|None|'] in help.md Input violations: Action -> "Generate AdHoc SQL Report": Missing ['|scope_ids|[]integer|[]|False|Scope IDs for which tickets should be generated, by default all are included|None|'] in help.md Input violations: Trigger -> "New Scans": Missing ["|status_filter|[]string|['Successful']|False|List of scan statuses to match for trigger; options include: Aborted, Successful, Running, Stopped, Failed, Paused, Unknown|None|"] in help.md Input violations: Trigger -> "New Vulnerability Exception": Missing ["|status_filter|[]string|['Under Review']|False|List of vulnerability statuses to match against. Options include: Under Review and Approved|None|"] in help.md`

@jfrantz1-r7 @jschipp-r7 
Help.md needs to be manually updated as there is a bug with `icon-plugin`. It is not being able to correctly generate `default` values. I would suggest to take latest copy of help.md from master and take backup of that file and then regenerate with icon-plugin. Overwrite the help.md with backup file and made your changes manually.

**Comment by zyoutz-r7** [comment]
It doesn't appear a `description` is necessary when using this endpoint; however, the `searchCriteria` is unfortunately.  This won't work:
```
{
	"name": "test",
	"type": "static"
}
```
But this will:
```
{
	"name": "test",
	"type": "static",
	"searchCriteria": {
		"match": "all",
		"filters": [
			{
				"field": "site-id",
                "operator": "in",
                "values": [1]
			}
		]
	}
}
```

**Comment by jfrantz1-r7** [comment]
@zyoutz-r7 that's weird, i had `searchCriteria` but no `description` and it wouldn't let me create it:

`Bad Request (400): The property 'description' must not be empty for the 'AssetGroup' object.
The property 'description' must not be empty for the 'AssetGroup' object.
Traceback (most recent call last):`

Using ICON:

image

**Comment by zyoutz-r7** [comment]
@jfrantz1-r7 This is because the plugin is still passing the `description` field with the payload with an empty string instead of not passing it at all.  Will need to do something similar to what was done for searchCriteria: https://github.com/rapid7/insightconnect-plugins/blob/master/rapid7_insightvm/komand_rapid7_insightvm/actions/create_asset_group/action.py#L21

So check if description is empty string, if so then pop from params so it isn't sent.
Screen Shot 2020-02-04 at 5 10 53 PM

**Pull request closed without merging**